server/update/router.ts

```diff
@@ -0,0 +1,154 @@
+import type {Request, Response, NextFunction} from 'express';
+import express from 'express';
+import UpdateCollection from './collection';
+import * as userValidator from '../user/middleware';
+import * as projectValidator from '../project/middleware';
+import * as updateValidator from './middleware';
+import * as util from './util';
+
+const router = express.Router();
+
+/**
+ * Get a given update by id
+ *
+ * @name GET /api/updates?updateId=updateId
+ *
+ * @return {UpdateResponse} - The update with id updateId
+ * @throws {400} - If updateId is not given
+ * @throws {403} - If user is not logged in or not part of the project the update is part of
+ * @throws {404} - If update with updateId is not found
+ */
+/**
+ * Get updates for a given project
+ *
+ * @name GET /api/updates?projectId=projectId
+ *
+ * @return {UpdateResponse[]} - An array of updates associated with the given project
+ * @throws {400} - If projectId is not given
+ * @throws {403} - If user is not logged in or not part of the project
+ * @throws {404} - If project with projectId is not found
+ */
+router.get(
+  '/',
+  [
+    userValidator.isUserLoggedIn,
+    projectValidator.isProjectExistsQuery,
+    projectValidator.isUserInProject,
+  ],
+  async (req: Request, res: Response, next: NextFunction) => {
+    // Check if updateId parameters was supplied instead
+    if (req.query.updateId !== undefined) { 
+      next();
+      return;
+    }
+
+    const updates = await UpdateCollection.findAllByProjectId(req.query.projectId as string);
+    const response = updates.map(util.constructUpdateResponse);
+    res.status(200).json(response);
+  },
+  [
+    updateValidator.isUserInProject,
+  ],
+  async (req: Request, res: Response) => {
+    const update = await UpdateCollection.findOneByUpdateId(req.query.updateId as string);
+    const response = util.constructUpdateResponse(update);
+    res.status(200).json(response);
+  },
+);
+
+/**
+ * Create a new update.
+ *
+ * @name POST /api/updates
+ *
+ * @param {string} status - The status of the update
+ * @param {string} summary - The summary of the update
+ * @param {string} details - The details of the update
+ * @param {string | undefined} todos - The todos of the update
+ * @param {string | undefined} blockers - The blockers of the update
+ * @param {string} projectId - The id of the project the update is associated with
+ * @return {UpdateResponse} - An object with the new update
+ * @throws {400} - If projectId is not given, or if update content is invalid
+ * @throws {403} - If user is not logged in or not part of the project
+ * @throws {404} - If project with projectId is not found
+ * @throws {413} - If summary exceeds 60 characters
+ */
+router.post(
```

Comment:
Just thought of something, but should we allow updates to projects that are inactive? Should we add a validator / prevent that?